encoding/da.go

```diff
@@ -714,3 +760,59 @@ func GetBatchEnableCompression(codecVersion CodecVersion, batch *Batch) (bool, e
 		return false, fmt.Errorf("unsupported codec version: %v", codecVersion)
 	}
 }
+
+func MessageQueueV2ApplyL1MessagesFromBlocks(initialQueueHash common.Hash, blocks []*Block) (common.Hash, error) {
+	rollingHash := initialQueueHash
+	for _, block := range blocks {
+		for _, txData := range block.Transactions {
+			if txData.Type != types.L1MessageTxType {
+				continue
+			}
+
+			data, err := hexutil.Decode(txData.Data)
+			if err != nil {
+				return common.Hash{}, fmt.Errorf("failed to decode txData.Data: data=%v, err=%w", txData.Data, err)
+			}
+
+			l1Message := &types.L1MessageTx{
+				QueueIndex: txData.Nonce,
+				Gas:        txData.Gas,
+				To:         txData.To,
+				Value:      txData.Value.ToInt(),
+				Data:       data,
+				// Sender:     , TODO: is this needed?
```

Comment:
Yes, needed, or the hashes won't match. This field is part of the [RLP encoding](https://github.com/scroll-tech/go-ethereum/blob/d2f19ff58eecb805ced826fc71da7521e9c2ea9b/core/types/l1_message_tx.go#L12).

```suggestion
				Sender:     txData.from,
```

Reply:
done in cbed8b2507be70f82e6cb4663668c56ed0c72edf